Testnet/StratisSwap/StratisSwap/SimpleSellOrder/SimpleSellOrder.cs

```diff
@@ -0,0 +1,212 @@
+using Stratis.SmartContracts;
+
+[Deploy]
+public class SimpleSellOrder : SmartContract
+{
+    /// <summary>
+    /// Constructor creating a simple sell order setting the token, price, and amount to sell.
+    /// </summary>
+    /// <param name="smartContractState">The execution state for the contract.</param>
+    /// <param name="token">The address of the src token being sold.</param>
+    /// <param name="price">The price for each src token in stratoshis.</param>
+    /// <param name="amount">The amount of src token to sell in full.</param>
+    public SimpleSellOrder(ISmartContractState smartContractState, Address token, ulong price, ulong amount)
+        : base (smartContractState)
+    {
+        Assert(price > 0, "Price must be greater than 0");
+        Assert(amount > 0, "Amount must be greater than 0");
+        Assert(PersistentState.IsContract(token), "Not a valid token address");
+
+        Token = token;
+        Price = price;
+        Amount = amount;
+        Seller = Message.Sender;
+        IsActive = true;
+    }
+
+    /// <summary>
+    /// The contract address of the token being sold.
+    /// </summary>
+    public Address Token
+    {
+        get => PersistentState.GetAddress(nameof(Token));
+        private set => PersistentState.SetAddress(nameof(Token), value);
+    }
+
+    /// <summary>
+    /// The price in stratoshis of each src token being sold.
+    /// </summary>
+    public ulong Price
+    {
+        get => PersistentState.GetUInt64(nameof(Price));
+        private set => PersistentState.SetUInt64(nameof(Price), value);
+    }
+
+    /// <summary>
+    /// The amount of src token being sold in full. 
+    /// </summary>
+    public ulong Amount
+    {
+        get => PersistentState.GetUInt64(nameof(Amount));
+        private set => PersistentState.SetUInt64(nameof(Amount), value);
+    }
+
+    /// <summary>
+    /// The seller wallet address.
+    /// </summary>
+    public Address Seller
+    {
+        get => PersistentState.GetAddress(nameof(Seller));
+        private set => PersistentState.SetAddress(nameof(Seller), value);
+    }
+
+    /// <summary>
+    /// Status flag for allowing/denying trades.
+    /// </summary>
+    public bool IsActive
+    {
+        get => PersistentState.GetBool(nameof(IsActive));
+        private set => PersistentState.SetBool(nameof(IsActive), value);
+    }
+
+    /// <summary>
+    /// Fully or partially fills a sell order.
+    /// </summary>
+    /// <param name="amountToBuy">The amount of SRC tokens to buy from the seller in full.</param>
+    /// <returns>Transaction result with trade details.</returns>
+    public Transaction Buy(ulong amountToBuy)
+    {
+        Assert(IsActive, "Contract is not active.");
+        Assert(Message.Sender != Seller, "Sender cannot be owner.");
+
+        amountToBuy = Amount >= amountToBuy ? amountToBuy : Amount;
```

Comment:
Good call, will fix this up.

Reply:
Doing this saves 152 gas but makes the code look very dirty. 152 gas * 100 price =  savings of .00152. I'm going to keep the code as is for now since it's not a critical amount. At $100 CRS is only saving 15 cents. Good to think about though for future work.